src/server/db_slice.h

```diff
@@ -497,6 +497,17 @@ class DbSlice {
   void PerformDeletion(Iterator del_it, DbTable* table);
   void PerformDeletion(PrimeIterator del_it, DbTable* table);
 
+  ConditionFlag* GetCondFlag() {
+    return &cond_flag_;
+  }
+
+  // Wrapper around DashTable::Traverse that allows preemptions
+  template <typename Cb, typename DashTable>
+  PrimeTable::Cursor Traverse(PrimeTable::Cursor cursor, Cb&& cb, DashTable* pt) {
```

Comment:
I'd have the callback be the last argument. Not sure how to rationalize it, but it seems more natural and common to me

Reply:
+1 :)